packages/frontend/src/Happy/Frontend.hs

```diff
@@ -0,0 +1,57 @@
+module Happy.Frontend where
+  
+import Happy.Grammar
+import Happy.Frontend.AbsSyn
+import Happy.Frontend.Mangler
+import Happy.Frontend.PrettyGrammar
+import Happy.Frontend.Parser
+import Happy.Frontend.ParseMonad.Class
+import System.Exit
+import System.IO
+
+parseYFileContents :: String -> ParseResult AbsSyn
+parseYFileContents contents = runFromStartP ourParser contents 1
+
+mangleAbsSyn :: AbsSyn -> String -> ParseResult Grammar
+mangleAbsSyn abssyn filename = first unlines (mangler filename abssyn)
+  where first f (Left a) = Left (f a)
+        first _ (Right b) = Right b
```

Comment:
This isn't really worth factoring out, it is just a little easy-to-write helper (especially with `Data.Bifunctor`). Better to keep that stuff as-is in the exe.